packages/core/src/indexing/database-index-api.ts

```diff
@@ -97,7 +123,10 @@ export abstract class DatabaseIndexApi<DateType = Date | number> {
         this.modelsToIndex.push(modelArgs.model)
       }
       if (modelArgs.relations) {
-        this.modelsIndexedFields.set(modelArgs.model.toString(), Object.keys(modelArgs.relations))
+        this.modelsIndexedFields.set(modelArgs.model, Object.keys(modelArgs.relations))
```

Comment:
this `modelsIndexedFields` member variable is confusingly named now.  It sounds like it could contain every field with an index on it, but it no longer does.  It only contains the fields which get duplicated out of the content JSON into their own column and have indices on those columns.  Right now that is only for fields that are relations.  We should make sure the distinction between any field with an index on the JSON content vs a relation field that is extracted out to its own column is clear.
We could call this `modelsExtractedFields` or even just `modelsRelationFields`, or something like that.  Probably should also add some comments explaining how this works